framework/doc/content/source/materials/CoupledValueFunctionMaterial.md

```diff
@@ -0,0 +1,14 @@
+# CoupledValueFunctionMaterial
+
+!syntax description /Materials/CoupledValueFunctionMaterial
+
```

Comment:
Does the capability introduced in this PR has to be necessarily used from a material?

Reply:
No, I modified ADBodyForce and added tests for it. That shows how to use it from a kernel.